### ISSUE TITLE
tramp mode presentation(s) ?

### ISSUE THREAD

**Issue by eddelbuettel** [ess-intro/ess-intro#14]
Tramp mode is good (tweet:  https://twitter.com/juxtaposz/status/1418648187599003649; screenshot below).  

We once talked about doing a presentation on it, or maybe even several.  

@chr1swallace how is https://github.com/ess-intro/presentation-remote-server coming along?

![image](https://user-images.githubusercontent.com/673121/126871497-78506e28-279d-48fa-a22c-dccbded6ec0a.png)

**Comment by chr1swallace** [comment]
I think it's short but complete. The beauty of tramp is that it basically just works so not much is needed. I haven't taken any steps into a screencast though which I guess would be the best demo.

https://chr1swallace.github.io


________________________________
From: Dirk Eddelbuettel ***@***.***>
Sent: Saturday, July 24, 2021 3:26:12 PM
To: ess-intro/ess-intro ***@***.***>
Cc: Chris Wallace ***@***.***>; Mention ***@***.***>
Subject: [ess-intro/ess-intro] tramp mode presentation(s) ? (#14)


Tramp mode is good (tweet: https://twitter.com/juxtaposz/status/1418648187599003649; screenshot below).

We once talked about doing a presentation on it, or maybe even several.

@chr1swallace<https://github.com/chr1swallace> how is https://github.com/ess-intro/presentation-remote-server coming along?

[image]<https://user-images.githubusercontent.com/673121/126871497-78506e28-279d-48fa-a22c-dccbded6ec0a.png>

**Comment by eddelbuettel** [comment]
Yes, a short video is the agreed-upon format. Many of us had good luck with [obs](https://obsproject.com/) to record.  If we can help, let us know.

**Comment by chr1swallace** [comment]
If you can get me to overcome embarrassment at hearing my own voice, then I'd gladly accept help! Otherwise I'll just have to be brave.

https://chr1swallace.github.io


________________________________

**Comment by eddelbuettel** [comment]
Yes, we all fight that fight.  I usually tell myself my own recordings don't exists which seems to work.  I then just nod politely when other people comment on them :stuck_out_tongue_winking_eye:

**Comment by bryce-carson** [comment]
I would love to have a better reference for LSP and TRAMP with R!

I have used Spacemacs for a good two years now, and I still haven't managed to get `company-mode`, `lsp-mode`, or any of that working (either as expected or at all) with ESS. `hippy-expand` is the only thing that works for me.

Enabling the `ess` and `lsp` layers in Spacemacs is not enough, apparently, for `company-mode` to work happily on local files. When _TRAMP_ is being used, it further complicates it. Registering a server and having the `languageserver.r` package installed on the remote host isn't enough too! Perhaps I did it wrong, but I'd love a _thorough_ literate configuration (a nice Rmd render?) to study.

For what it is worth, here's what I have in my `.spacemacs`. What do you have?
```elisp
;; Inside layers
     lsp
     (ess :variables
          ess-r-backend 'lsp
          ess-use-company 'script-only)
```

```elisp
;; Inside my user-config
  (customize-set-variable 'tramp-remote-path '(tramp-default-remote-path "~/bin" "~/" "~/R"))
  (with-eval-after-load 'lsp-mode
    (lsp-register-client
     (make-lsp-client :new-connection (lsp-tramp-connection "lsp-r")
                      :major-modes '(ess-r-mode)
                      :remote? t
                      :server-id 'lsp-r)))
```

I wish my experience were that as pictured in the Tweet.

**Comment by chr1swallace** [comment]
I don't use spacemacs but doom.

I don't have anything configured on the remote host.  I load tramp with

(use-package! tramp)
(after! tramp
  (setq tramp-default-method "ssh")
  (add-to-list 'tramp-remote-path 'tramp-own-remote-path))
(use-package! ess)

(use-package! polymode
  :ensure t
  :mode ("\\.[rR]md\\'" . poly-markdown+R-mode))
(use-package! poly-R
  :after polymode)

But I also don't use company, it felt slow over tramp.  I use
vertico https://github.com/minad/vertico and this seems to work (as
much as I want it to) out of the box.  That is, I can expand in the *R*
buffer, but not in the script buffer.  I don't know how to get that
working.

Not sure that helps!

**Comment by bryce-carson** [comment]
Thanks for the reply. I have essentially the same setup in Spacemacs, but I'll take a look at Vertico and see what it offers.

I'll talk to the other ESS contributors to see what they can advise, and the Spacemacs people as well.

With that done in the future, I'll make another issue and we could write a more complete document for the EES intro series about both TRAMP and autocompletion.

**Comment by chr1swallace** [comment]
that would be great.
